src/inputconnectorstrategy.h

```diff
@@ -216,12 +217,33 @@ namespace dd
         }
     }
 
+    // TODO: inheritance between pred dto & train dto
+    void get_data(oatpp::Object<DTO::ServicePredict> pred_dto)
+    {
+      _uris.clear();
+      for (auto &uri : *pred_dto->data)
+        _uris.push_back(uri->std_str());
+      _ids = pred_dto->_ids;
+      _meta_uris = pred_dto->_meta_uris;
+      _index_uris = pred_dto->_index_uris;
+
+      if (_uris.empty())
+        {
+          throw InputConnectorBadParamException("missing data");
+        }
+    }
+
     void set_timeout(const APIData &ad)
```

Comment:
Do we want to mark these functions as deprecated, for later removal ?

Reply:
This function is still used by the APIData parts of the code atm, that's why it's still there